dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/deduplication/DeduplicationControllerTest.java

```diff
@@ -0,0 +1,169 @@
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.deduplication;
+
+import static org.hisp.dhis.web.WebClientUtils.assertStatus;
+
+import org.hisp.dhis.common.CodeGenerator;
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.deduplication.DeduplicationStatus;
+import org.hisp.dhis.deduplication.PotentialDuplicate;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.trackedentity.TrackedEntityInstance;
+import org.hisp.dhis.web.HttpStatus;
+import org.hisp.dhis.webapi.DhisControllerConvenienceTest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * @author luca@dhis2.org
+ */
+class DeduplicationControllerTest extends DhisControllerConvenienceTest
+{
+    private final static String ENDPOINT = "/" + "potentialDuplicates/";
+
+    @Autowired
+    private IdentifiableObjectManager dbmsManager;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private TrackedEntityInstance origin;
+
+    private TrackedEntityInstance duplicate;
+
+    @BeforeEach
+    public void setUp()
+    {
+        OrganisationUnit orgUnit = createOrganisationUnit( CodeGenerator.generateUid() );
+        dbmsManager.save( orgUnit );
+
+        origin = createTrackedEntityInstance( orgUnit );
+        duplicate = createTrackedEntityInstance( orgUnit );
+
+        dbmsManager.save( origin );
+        dbmsManager.save( duplicate );
+    }
+
+    @Test
+    void shouldPostPotentialDuplicateWhenTrackedEntitiesExist()
+        throws Exception
+    {
+        PotentialDuplicate potentialDuplicate = new PotentialDuplicate( origin.getUid(), duplicate.getUid() );
+        assertStatus( HttpStatus.OK, POST( ENDPOINT, objectMapper.writeValueAsString( potentialDuplicate ) ) );
+    }
+
+    @Test
+    void shouldThrowPostPotentialDuplicateWhenMissingDuplicateTeiInPayload()
+        throws Exception
+    {
+        PotentialDuplicate potentialDuplicate = new PotentialDuplicate( origin.getUid(), null );
+        assertStatus( HttpStatus.BAD_REQUEST, POST( ENDPOINT, objectMapper.writeValueAsString( potentialDuplicate ) ) );
```

Comment:
We have multiple tests that are testing the same endpoint and method. Since we only assert on the status code there is a risk that multiple tests test the same case. Even if this is not the case now it could be after a refactoring and not be caught by the tests.

I wonder if there is a keyword specific to these cases that we can assert on in the body? Not asserting on the exact message.